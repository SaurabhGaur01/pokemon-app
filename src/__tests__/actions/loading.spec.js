import {
    changeLoading,
    loadingReducer,
} from '../../actions/loading';

describe('loading Tests', () => {
    describe('Action Generator', () => {
        it('setDrawerClosed should match the snapshot' , () => {
            expect(changeLoading(true)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(loadingReducer()).toEqual(true);
        });
        it('should return the value to true when true flag is passed' , () => {
            expect(loadingReducer('', changeLoading(true))).toEqual(true);
        });
        it('should return the value to false when false flag is passed' , () => {
            expect(loadingReducer('', changeLoading(false))).toEqual(false);
        });
    });
});