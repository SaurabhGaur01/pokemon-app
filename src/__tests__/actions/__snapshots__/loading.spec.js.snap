// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading Tests Action Generator setDrawerClosed should match the snapshot 1`] = `
Object {
  "isLoading": true,
  "type": "CHANGE_LOAD",
}
`;
